orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -159,8 +162,11 @@ class BakeStage implements StageDefinitionBuilder {
       ]
 
       if (failOnImageNameMismatchEnabled()) {
-         List<String> distinctImageNames = globalContext.deploymentDetails.stream()
-          .map({details -> details['imageName']})
+        // find distinct image names in bake stages that are actually related to the stage passed into the task
+        def distinctImageNames = relatedBakeStages
+          .findAll { it.parentStageId == stage.id && (it.context.ami || it.context.imageId) }
```

Comment:
is the `(it.context.ami || it.context.imageId)` to determine whether this is a bake stage?

would checking it.type == 'bake' accomplish this instead?

alternatively would `it.context.imageName` be more appropriate to check since we are mapping to that on L169?

Reply:
I was following what was done here:
https://github.com/spinnaker/orca/blob/608099016ea9436a032a88487ffd296b7bdbd350/orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy#L149
...which I guessed was filtering for stages that actually had bake output information in them (without `ami` or `imageId` in the context, the bake probably failed?).

Does that make sense? Maybe I could encapsulate that in a method like `hasValidBakeContext(stage)` to make that clearer and reusable?

Reply:
then it's probably more appropriate to use:
```suggestion
          .findAll { it.parentStageId == stage.id && (it.context.imageName) }
```

since that's what you actually use for comparison on L169

Reply:
@marchello2000 Am I guaranteed to have `imageName` in the context if the bake succeeded?

Reply:
Changed it to use `imageName`. Better?